byway/src/main/java/com/google/sps/servlets/StopsServlet.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/stop")
+public class StopsServlet extends HttpServlet {
+  ArrayList<String> stops= new ArrayList<String>();
```

Comment:
There's only one instance of a given Servlet per server, so whatever data you put in `stops` will be shared across all calls to doGet and doPost for all users. This will get even more complicated when you move to deploy this on GCP as GCP may run 0 to N copies of your server depending on the amount of traffic you are getting, which means that a GCP server restart could unexpectedly clear this list, or if GCP is running two instances of your server, there will be two copies of this list with different data (which is nonetheless shared between multiple users).

Additionally, Servlets use multi-threading to handle simultaneous requests, so you could be doing a thread-unsafe sumultaneous write if multiple calls to doPost come in at the same time from different users. Try just opening your website in multiple tabs (or deploy it and try having everyone try to use this do-post action simultaneously multiple times in a row), and see what happens.

Reply:
I was just using arraylists for hardcoded values but I did add in datastore, but still creating entries w hardcoded values.

Reply:
For temporary usage like this, it would still be good to use something like [`@GuardedBy("this")`](https://errorprone.info/bugpattern/GuardedBy), and make the request handlers `synchronized`, then.

Also make it private and final if you don't need public access or to assign new values later.

Reply:
Forgot to comment, done!